mantidproject/mantid
# Add new SetMDFrames algorithm

Fixes #14239 

OSX and Windows installers from the last Jenkins build can be found here: \olympic\Babylon5\Scratch\Anton\Testing\14239_SetMDFrames
# For testing
1. Code review
2. Check that the documentation in the rst file is understandable.
3. Open the SetMDFrames dialog and check that it looks alright
### Test instructions

The tests are based on the following script

``` python
# Prepare the sample workspace
inputFrame = 'QSample,QSample,QSample'
inputUnits = 'U,U,U'
outputFrame = 'QLab'
mdws = CreateMDWorkspace(Dimensions=3, Extents='-10,10,-10,10,-10,10', 
                                        Names='A,B,C', Units=inputUnits, 
                                        Frames=inputFrame)
print "The first MDFrame is of type " + mdws.getDimension(0).getMDFrame().name()
print "The second MDFrame is of type " + mdws.getDimension(0).getMDFrame().name()
print "The third MDFrame is of type " + mdws.getDimension(0).getMDFrame().name()

 # Set the MDFrames to new values
SetMDFrame(mdws, MDFrame=outputFrame, Axes=[0,2])
print
print "The new first MDFrame is of type " + mdws.getDimension(0).getMDFrame().name()
print "The second MDFrame is of type " + mdws.getDimension(1).getMDFrame().name() 
print "The new second MDFrame is of type " + mdws.getDimension(2).getMDFrame().name()
```

These tests are similar to the unit tests, but might give a starting point for random testing

**Test regular input and output_**
1. Set the input Frame to 'QSample,QSample,QSample'
2. Set the input Units to 'U,U,U'
3. Set the output frame to 'QLab'
4. Run the script
   - Confirm that QLab frame was chosen (should be printed out) for index 0 and index 2

**Test HKL selection with bad units**
1. Set the input Frame to 'General Frame,General Frame,General Frame'
2. Set the input Units to 'isdfsdf, gg, sdfdf'
3. Set the output frame to "HKL"
4. Run the script
   - Confirm that an error is generated

**Test HKL selection with bad units**
1. Set the input Frame to  'General Frame,General Frame,General Frame'
2. Set the input Units to 'in 2.6437 A^-1,in 2.6437 A^-1,in 2.6437 A^-1'
3. Set the output frame to "HKL"
4. Run the script
   - Confirm that the output frame is HKL (should be printed out) for index 0 and index 2

**Test incorrect selection**
1. Set the input Frame to 'General Frame,General Frame,General Frame'
2. Set the input Units to  'in 2.6437 A^-1,in 2.6437 A^-1,in 2.6437 A^-1'
3. Set the output frame to "sdfsdf"
4. Run the script
   - Confirm that an error is generated
